internal/network/resource_network_peer.go

```diff
@@ -0,0 +1,391 @@
+package network
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	lxd "github.com/canonical/lxd/client"
+	"github.com/canonical/lxd/shared/api"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/terraform-lxd/terraform-provider-lxd/internal/common"
+	"github.com/terraform-lxd/terraform-provider-lxd/internal/errors"
+	provider_config "github.com/terraform-lxd/terraform-provider-lxd/internal/provider-config"
+)
+
+// NetworkPeerModel is a resource data model that matches the schema.
+type NetworkPeerModel struct {
+	Name        types.String `tfsdk:"name"`
+	Description types.String `tfsdk:"description"`
+
+	// Source network.
+	SourceNetwork types.String `tfsdk:"source_network"`
+	SourceProject types.String `tfsdk:"source_project"`
+
+	// Target network.
+	TargetNetwork types.String `tfsdk:"target_network"`
+	TargetProject types.String `tfsdk:"target_project"`
+
+	Remote types.String `tfsdk:"remote"`
+	Config types.Map    `tfsdk:"config"`
+	Status types.String `tfsdk:"status"`
+}
+
+// NetworkPeerResource represent LXD network peer resource.
+type NetworkPeerResource struct {
+	provider *provider_config.LxdProviderConfig
+}
+
+// NewNetworkPeerResource returns a new network peer resource.
+func NewNetworkPeerResource() resource.Resource {
+	return &NetworkPeerResource{}
+}
+
+// Metadata for network peer resource.
+func (r NetworkPeerResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = fmt.Sprintf("%s_network_peer", req.ProviderTypeName)
+}
+
+// Schema for network peer resource.
+func (r NetworkPeerResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				Description: "Name of the network peer",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
```

Comment:
Seems weird, but they are aligned. Fields `Description` and `Required` are aligned, but `PlanModifiers` is not because its value spans across multiple lines.

Reply:
For example, if PlanModifiers would be in a single line:
```go
"name": schema.StringAttribute{
	Description:   "Name of the network peer",
	Required:      true,
	PlanModifiers: []planmodifier.String{stringplanmodifier.RequiresReplace()},
	Validators: []validator.String{ // Notice that here is not aligned due to "multi-line" value.
		stringvalidator.LengthAtLeast(1),
	},
},
```

Reply:
Hmm, how strange, alrighty.